gen/templates/generic/data_source.go

```diff
@@ -152,18 +158,37 @@ func (d *{{camelCase .Name}}DataSource) Read(ctx context.Context, req datasource
 		return
 	}
 
-	tflog.Debug(ctx, fmt.Sprintf("%s: Beginning Read", config.Id.String()))
 
-	res, err := d.client.Get("{{if .GetRestEndpoint}}{{.GetRestEndpoint}}{{else}}{{.RestEndpoint}}{{end}}" + config.Id.ValueString())
+	{{ if not .RemoveId}}
+	tflog.Debug(ctx, fmt.Sprintf("%s: Beginning Read", config.Id.String()))
+	{{- end}}
+	{{if .RemoveId}}
+
+
+		var params = "?"
+		{{range $i, $a := .Attributes}}
+		{{- if not .Value}}
+		{{- if .QueryParam}}
+		if(!config.{{toGoName .TfName}}.IsNull()) {
+			params = params + "{{.ModelName}}=" + config.{{toGoName .TfName}}.ValueString() + "&"
```

Comment:
Do we need to consider url encoding here?

Reply:
We might but I dont know if we do for these attributes. Serial Number may have impact with that